allure-testng-adaptor/src/main/java/ru/yandex/qatools/allure/testng/AllureTestListener.java

```diff
@@ -233,4 +244,87 @@ String getSuiteUid(ITestContext iTestContext) {
         }
         return uid;
     }
+
+    /**
+     * Creates test case parameters in XML in case of parametrized test (see TestNG @DataProvider).
+     * Java 8 allows to keep test method parameter names in class files (needs javac -parameters compilation).
+     * In Java 7 default parameter naming is used (arg0, arg1, ... ).
+     *
+     * @param iTestResult parameter passed to onTestStart() callback by TestNG
+     */
+    private void fireAddParameterEvents(ITestResult iTestResult) {
+        Object[] parameters = iTestResult.getParameters();
+        ITestNGMethod method = iTestResult.getMethod();
+        Method nativeMethod = method.getConstructorOrMethod().getMethod();
+        Annotation[][] parametersAnnotations = nativeMethod.getParameterAnnotations();
+
+        boolean atLeastJava8 = isAtLeastJava8();
+        List<String> parameterNames = new ArrayList<>();
+        if (atLeastJava8) {
+            parameterNames = getParameterNames(nativeMethod);
+        }
+
+        for (int i = 0; i < parameters.length; i++) {
+
+            // searching for annotated parameters
+            Annotation[] parameterAnnotations = parametersAnnotations[i];
```

Comment:
better to use method like `getAnnotationForParameter(Annotation[][] annotations, int parameterIndex)`